pkgs/fc/sensuplugins/fc/sensuplugins/http.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+"""HTTP status check.
+
+Check the status code from an HTTP request, with options for following
+redirects or not.
+"""
+
+import argparse
+import os
+import sys
+import urllib.parse
+
+import requests
+from requests_toolbelt.adapters.source import SourceAddressAdapter
+
+
+def err(msg):
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+
+def unknown(msg):
+    print(f"HTTP UNKNOWN - {msg}")
+    sys.exit(1)
+
+
+def warning(msg):
+    print(f"HTTP WARNING - {msg}")
+    sys.exit(1)
+
+
+def critical(msg):
+    print(f"HTTP CRITICAL - {msg}")
+    sys.exit(1)
+
+
+def ok(msg):
+    print(f"HTTP OK - {msg}")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-4", "--ipv4", action="store_true", help="Only use IPv4"
+    )
+    parser.add_argument(
+        "-6", "--ipv6", action="store_true", help="Only use IPv6"
+    )
+    parser.add_argument(
+        "-e",
+        "--expect",
+        type=int,
+        action="append",
+        help="Expected status codes",
+    )
+    parser.add_argument(
+        "-f", "--follow", action="store_true", help="Follow HTTP redirects"
+    )
+    parser.add_argument("URL", help="URL to be fetched")
+
+    args = parser.parse_args()
+
+    # forcing a specific address family by binding to its wildcard
+    # address works on linux, but does not seem to work on darwin.
+    source_address = None
+    if args.ipv4 and args.ipv6:
+        err("conflicting arguments, -4 and -6 cannot be specified together")
+    elif args.ipv4:
+        source_address = "0.0.0.0"
+    elif args.ipv6:
+        source_address = "::"
+
+    if args.expect is not None and any(
+        map(lambda n: n < 100 or n > 599, args.expect)
+    ):
+        err("invalid expected status code")
+
+    try:
+        url = urllib.parse.urlsplit(args.URL)
+    except ValueError as ex:
+        err(f"could not parse URL: {ex}")
+
+    if url.scheme != "http" and url.scheme != "https":
+        err(f"unsupported url scheme: {url.scheme}")
+
+    if url.hostname is None:
+        err("empty hostname in URL")
+
+    session = requests.Session()
+    # match monitoring-plugins defaults
+    session.max_redirects = 10
+
+    if url.username is not None:
+        session.auth = (url.username, url.password)
+
+    target = urllib.parse.urlunsplit(
+        (url.scheme, url.hostname, url.path, url.query, "")
+    )
+
+    if source_address is not None:
+        session.mount("http://", SourceAddressAdapter(source_address))
+        session.mount("https://", SourceAddressAdapter(source_address))
+
+    try:
+        resp = session.get(target, allow_redirects=args.follow)
+    except requests.ConnectionError as ex:
+        critical("could not connect to remote host")
```

Comment:
adding exception detail here would be great, or at least the specific connection error message

Reply:
Fixed in [39ac04b](https://github.com/flyingcircusio/fc-nixos/pull/940/commits/39ac04b0256307c4a9dc61bd0c4b9d08e9284798).